agent_testing/agent.py

```diff
@@ -127,6 +127,90 @@ async def delete_item(self, message, conn):
+        await conn.send_async(
+            {
+                "@type": self.type("message"),
+                "~l10n": {"locale": "en"},
+                "sent_time": datetime.now().strftime("%Y-%m-%d %H:%M:%S"),
+                "content": "Blockchain is a type of database that stores "
+                "information in a decentralized, distributed digital ledger.",
+            }
+        )
+
+    @route("https://example.com/message-exchange/0.1/message2")
+    async def second_message(self, msg, conn):
+        """Send and receive second messages."""
+        await conn.send_async(
+            {
+                "@type": self.type("message"),
+                "~l10n": {"locale": "en"},
+                "sent_time": datetime.now().strftime("%Y-%m-%d %H:%M:%S"),
+                "content": "Data is recorded in a way that is transparent, "
+                "time stamped, and immutable. (Type one of the previous "
+                "words to learn more.)",
+            }
+        )
+
+    @route("https://example.com/message-exchange/0.1/message3")
+    async def third_message(self, msg, conn):
+        """Send and receive third messages."""
+        if "transparent" == msg["item"].lower():
+            await conn.send_async(
+                {
+                    "@type": self.type("message"),
+                    "~l10n": {"locale": "en"},
+                    "sent_time": datetime.now().strftime("%Y-%m-%d %H:%M:%S"),
+                    "content": "This means that everyone can see that there "
+                    "was a transaction.",
+                }
+            )
+        elif "time stamped" == msg["item"].lower():
+            await conn.send_async(
+                {
+                    "@type": self.type("message"),
+                    "~l10n": {"locale": "en"},
+                    "sent_time": datetime.now().strftime("%Y-%m-%d %H:%M:%S"),
+                    "content": "This means that everyone can see when the "
+                    "transaction was made.",
+                }
+            )
+        elif "immutable" == msg["item"].lower():
+            await conn.send_async(
+                {
+                    "@type": self.type("message"),
+                    "~l10n": {"locale": "en"},
+                    "sent_time": datetime.now().strftime("%Y-%m-%d %H:%M:%S"),
+                    "content": "This means that no one can alter previous "
+                    "transactions or information on the ledger.",
+                }
+            )
+        else:
+            await conn.send_async(
+                {
+                    "@type": self.type("message"),
+                    "~l10n": {"locale": "en"},
+                    "sent_time": datetime.now().strftime("%Y-%m-%d %H:%M:%S"),
+                    "content": "Sorry, we didn't understand your request! "
+                    "Please type 'transparent', 'time stamped', 'immutable' "
+                    "(not case sensitive).",
+                }
+            )
+
+
```

Comment:
Nice work implementing the branching paths of this message! Here's a recommendation for one possible optimization.

When we see a lot of code that looks very similar with minor mutations, we should think of loops, data structures, and methods to maximize our code reuse :slightly_smiling_face: 

For this one, the message sending and several of the attributes in the message look the same for all paths through the tree. We can make that clearer by just setting the `content` as a result of the branches and taking one action at the end instead of taking the action in each branch.

For example:
```python
if choice == "one":
    content = "content of choice one"
elif choice == "two":
    content = "content of choice two"
else:
    content = "you didn't choose one or two"

my_action(content)
```

Another possible simplification would be to use a dictionary to record our responses to certain inputs:

```python
responses = {
    "one": "content of choice one",
    "two": "content of choice two"
}
content = responses.get(choice, "you didn't chose one or two")
my_action(content)
```